### PR TITLE
kubernetes CI pipeline

### PR DESCRIPTION
**Pull request by matyasfodor**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      REPO: "dddecaf/decaf-metabolica"
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run: npm install
+      - run: ./.circleci/environment.sh npm run build
+      - run: ./.circleci/install_gcloud.sh
+      - run: ./.circleci/install_kubectl.sh
+      - run: docker build -f Dockerfile -t $REPO:$CIRCLE_BRANCH .
+      - run: docker tag $REPO:$CIRCLE_BRANCH $REPO:${CIRCLE_SHA1::12}
```
Comment (kvikshaug) [inline .circleci/config.yml]: FYI you can add all tags while building, f.ex. `docker build -t $REPO:$CIRCLE_BRANCH -t $REPO:${CIRCLE_SHA1::12} .`

Comment (kvikshaug) [inline .circleci/config.yml]: Missing a push step here, both tags should be pushed

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/node:9.11.1
+    environment:
+      REPO: "dddecaf/decaf-metabolica"
```
Comment (kvikshaug) [inline .circleci/config.yml]: The repo should be `gcr.io/dd-decaf-cfbf6/decaf-metabolica` as we are moving our images over to GCR where we have unlimited private repos.

File: .circleci/push_to_hub.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+set -ev
+REPO="dddecaf/decaf-metabolica"
+GIT_SHA=$(git rev-parse --short=12 --verify HEAD)
+docker build -f Dockerfile -t $REPO:$CIRCLE_BRANCH .
+docker tag $REPO:$CIRCLE_BRANCH $REPO:$GIT_MASTER_HEAD_SHA
+docker push $REPO:$CIRCLE_BRANCH
+docker push $REPO:$GIT_MASTER_HEAD_SHA
```
Comment (kvikshaug) [inline .circleci/push_to_hub.sh]: This script can be deleted as the build/tag/push logic is in circleci config now

File: Dockerfile [hunk 4]
```diff
@@ -1,6 +1,4 @@
-FROM alpine:latest
-RUN mkdir -p /usr/share/decaf/frontend
-ADD . /usr/share/decaf/frontend
-WORKDIR /usr/share/decaf/frontend
-
-RUN chmod -R 755 /usr/share/decaf/frontend/dist
+FROM nginx:1.13
+# RUN mkdir -p /usr/share/decaf/frontend
```
Comment (kvikshaug) [inline Dockerfile]: Remove commented line

File: nginx.conf [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        gzip_types text/plain application/xml;
+
+        location / {
+            include  /etc/nginx/mime.types;
+
+            root /usr/share/nginx/html;
+            try_files $uri /index.html =404;
+        }
+    }
+}
```
Comment (kvikshaug) [inline nginx.conf]: Missing newline at EOF

File: package.json [hunk 6]
```diff
@@ -10,7 +10,7 @@
     "metabolica-core": "github:biosustain/metabolica-ui-core",
     "metabolica-home": "github:DD-DeCaF/metabolica-ui-home#v1.0.2",
     "metabolica-login": "github:DD-DeCaF/metabolica-ui-login#v1.0.2",
-    "metabolica-map": "github:DD-DeCaF/metabolica-ui-map#v1.2.1",
+    "metabolica-map": "github:DD-DeCaF/metabolica-ui-map#devel",
```
Comment (kvikshaug) [inline package.json]: Please don't merge devel dependencies into master!

**Review by kvikshaug — COMMENTED** [review]
This looks good, just a few comments. It's possible to cache dependencies to speed up `npm install` but I don't think that's very urgent - maybe better to wait until we're on Travis so we don't need to solve it twice.

File: package.json (hunk 6 above)
Comment (matyasfodor) [reply package.json]: This one is not perfectly figured out. On devel branch we are referencing the devel dependencies, on master the versioned ones. I'm not sure if it's possible to dispatch this in package.json

**Review by kvikshaug — APPROVED** [review]
LGTM

**Pull request merged**